src/shell/lean.cpp

```diff
@@ -503,8 +503,9 @@ int main(int argc, char ** argv) {
             }
         }
 
+        // Options appear to be empty, pretty sure I'm making a mistake here.
         if (compile && !mods.empty()) {
-            auto opts = ios.get_options();
+            auto opts = mod_mgr.get_options(); // ios.get_options();
```

Comment:
`mod_mgr.get_options()` just returns the initial options, not the options obtained after parsing a module (the method does not even know which module we're interested in).  At the moment we do not return the options from module_mgr, but you can just add a field to `module_info::parse_result`, right next to `m_env`.